### PR TITLE
JDK 11 support

### PR DESCRIPTION
**Pull request by jdubois**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by cbornet** [comment]
Isn't this going to conflict with the Spring-Boot 2.1 migration ?

**Comment by jdubois** [comment]
Normally not, I'm doing this for the Spring Boot 2.1 migration in fact (I'm going to merge this in the 2.1 branch)

**Comment by cbornet** [comment]
ok

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -54,7 +54,6 @@ plugins {
 apply plugin: 'java'
 sourceCompatibility=<%= JAVA_VERSION %>
 targetCompatibility=<%= JAVA_VERSION %>
-// Until JHipster supports JDK 9
 assert System.properties['java.specification.version'] == '1.8'
```
Comment (xetys) [inline generators/server/templates/build.gradle.ejs]: isn't this line preventing a project to built by gradle, if you try this with JDK 11?

**Comment by DanielFran** [comment]
@jdubois In my side I was adding:
		dependency 'com.sun.xml.bind:jaxb-core:2.3.0.1
		dependency 'com.sun.xml.bind:jaxb-impl:2.3.0.1
		dependency 'javax.xml.bind:jaxb-api:2.4.0-b180830.0359

File: generators/server/templates/build.gradle.ejs (hunk 1 above)
Comment (jdubois) [reply generators/server/templates/build.gradle.ejs]: Oh you are correct!!!! Let me have a look

Comment (jdubois) [reply generators/server/templates/build.gradle.ejs]: OK I'm fixing this in 2 minutes

**Comment by jdubois** [comment]
Sorry everyone, I messed up here.

In fact this can only work with Spring Boot 2.1 -> I'm reverting my changes and pushing them only to the 2.1 branch

**Comment by DanielFran** [comment]
@jdubois Should we define it in jhipster project and only declare de dependencies without version in the generator?

**Comment by jdubois** [comment]
@DanielFran for me this is a small hack that should be temporary, let's only do this in the generator

**Comment by jdubois** [comment]
Done in the `spring-boot_2.1.0` branch

**Pull request merged**